### PR TITLE
fix: make sure that we use a sanitized version of the git remote URL

### PR DESCRIPTION
**Pull request by metacosm**
Also make sure that the remote URL rewriting works for gitlab as well.

### CODE REVIEW

**Comment by cmoulliard** [comment]
LGTM. To be tested of course

**Comment by aureamunoz** [comment]
LGTM. I'm wondering if it would make sense to add a check in the `Issue52Test` in order to validate that `component.getSpec().getBuildConfig().getUrl()` is as expected (I don't know right now if it could break the build during release or in a CI context). WDYT @cmoulliard @metacosm ?

**Comment by cmoulliard** [comment]
> WDYT

This is an excellent idea. You can use a dummy github project from which you know what should be the URL address and then check it @aureamunoz

**Comment by metacosm** [comment]
We could add more tests but there's already unit test coverage.

**Comment by geoand** [comment]
Is suppose this resolves #310?

File: core/src/test/java/io/dekorate/utils/GitTest.java [hunk 1]
```diff
@@ -1,74 +1,126 @@
...
-
-  @Test
-  public void shouldGetUrl() throws Exception {
-    Optional<String> repoUrl = Git.getRemoteUrl(ROOT, Git.ORIGIN);
+  
+  private static Path getRoot(File configFile) {
+    return configFile.toPath().getParent().getParent();
+  }
+  
+  public static void setup(String gitConfig) {
```
Comment (geoand) [inline core/src/test/java/io/dekorate/utils/GitTest.java]: This and the methods below don't have to be `public`. Maybe they should be package-private like the rest?

**Comment by geoand** [comment]
I don't think we need an e2e for this. The feature seems very well covered by unit tests.

**Pull request merged**